File: game-engine/src/main/java/com/animousen4/game/engine/core/dao/UserDao.java
package com.animousen4.game.engine.core.dao;

import com.animousen4.game.engine.core.repositories.UserRepository;
import com.animousen4.game.engine.core.repositories.entities.UserEntity;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class UserDao{

    @Autowired
    UserRepository userRepository;

    public UserEntity getUserById(Long id) {
        return userRepository.findUserEntityById(id);
    }
    public UserEntity getUserByUsername(String username) {
        return userRepository.findUserEntityByUsername(username);
    }
    public void updateUserById(UserEntity user) {
        userRepository.save(user);
    }
}

File: auth-server-webflux/src/main/kotlin/com/animousen4/authserverwebflux/core/dao/UserDao.kt
package com.animousen4.authserverwebflux.core.dao

import com.animousen4.authserverwebflux.core.entities.UserEntity
import com.animousen4.authserverwebflux.core.repo.UserRepository
import org.springframework.stereotype.Component
import org.springframework.stereotype.Service
import reactor.core.publisher.Mono

@Component
class UserDao(
    val userRepository: UserRepository
) {
    fun getUserByEmail(email: String) = userRepository.getUserEntityByEmail(email)
}